jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -95,6 +96,20 @@ private List<TarArchiveEntry> getSortedEntries() {
     }
   }
 
+  private static void clearTimeHeaders(TarArchiveEntry entry, Instant modTime) {
+    entry.setModTime(modTime.toEpochMilli());
+
+    String headerTime = Long.toString(modTime.getEpochSecond());
+    final long nanos = modTime.getNano();
```

Comment:
I'm not familiar with the PAX headers, so just asking: does it have to be the nanosecond precision? Like, what if someone decides to go with milli or microseconds? Is it possible? Not that I care as long as nano works.

Reply:
Here is a post I was able to find:

> This is important for example with the tarfile module with the pax tar format. The POSIX tar [standard](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/pax.html) mandates storing the mtime in the extended header (if it is not an integer) with as much precision as is available in the underlying file system, and likewise to restore this time properly upon extraction.

So I'd say that it's better to store time with nanosecond precision. File systems that don't support this precision won't fail upon extraction: they'll simply truncate values.